keras/backend/tensorflow_backend.py

```diff
@@ -1239,8 +1242,8 @@ def rnn(step_function, inputs, initial_states,
                 # That's what the tile call does,
                 # it just repeats the mask along its second dimension
                 # n times.
-                tiled_mask_t = tf.tile(mask_t,
-                                       tf.pack([1, tf.shape(output)[1]]))
+                mask_t = tf.squeeze(mask_t)
+                tiled_mask_t = mask_t
```

Comment:
Sorry about that, `condition` in `tf.select` support a vector shape which is not very same with the compatible behavior in `T.switch` and the comment should be updated. 
If we use `tf.tile` to make `mask.shape == x.shape`, I think the code would be redundant and `tf.squeeze` would be a suitable choice.